blockchain/fullblocktests/generate.go

```diff
@@ -1089,6 +1092,95 @@ func Generate(includeLargeReorg bool) (tests [][]TestInstance, err error) {
 	})
 	rejected(blockchain.ErrFreshStakeMismatch)
 
+	// Attempt to add block with tickets voting on the wrong block
+	//
+	//   ... -> b36(8)
+	//                \-> bv6(9)
+	g.SetTip("b36")
+	g.NextBlock("bv6", outs[9], ticketOuts[9], func(b *wire.MsgBlock) {
+		// invalidate the vote by corrupting the output script
+		b.STransactions[0].TxOut[0].PkScript[2] &= 0x00
```

Comment:
This is not reliable because the block being voted on could already have a zero there in which case the test would fail with a false positive.

You need to ensure some bits are flipped.  So, I would suggest using xor.  You could do `b.STransactions[0].TxOut[0].PkScript[2] ^= 0x55` to flip every other bit in the byte.